stdlib/dynarray.mli

```diff
@@ -308,6 +308,31 @@ val filter_map : ('a -> 'b option) -> 'a t -> 'b t
     ignoring strings that cannot be converted to integers.
 *)
 
+(** {1:comparison Comparison functions}
+
+    Comparison functions iterate over their two dynarray inputs; it is
+    a programming error to change their length during the iteration,
+    see the {{!section:iteration} Iteration} section above.
+ *)
+
+val equal : ('a -> 'a -> bool) -> 'a t -> 'a t -> bool
+(** [equal eq a b] holds when [a] and [b] have the same length,
+    and for each pair of elements [ai], [bi] at the same
+    position we have [eq ai bi].
```

Comment:
Note that position has a specific meaning in OCaml (see the `String` module). I would rather use index (there's a few places in the `.mli` where index should also be substituted in my opinion). Also I would simply use `get` to index: 

```
(** [equal eq a b] holds when [a] and [b] have the same length,
    and for all indices [i] we have [eq (get a i) (get b i)]. *)
```